### PR TITLE
Extract new plugin-home-react package and deprecate createCardExtension in plugin-home

### PR DESCRIPTION
**Pull request by malikahmontgomery**
## Hey, I just made a Pull Request!
Closes https://github.com/backstage/backstage/issues/17737

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/home-react/package.json
Comment (awanlin) [inline plugins/home-react/package.json:4]: You'll want to leave this as `0.0.0` and then it will get bumped when it gets release for the first time

File: plugins/home-react/README.md
Comment (awanlin) [inline plugins/home-react/README.md:1]: I'm not sure if all of this should be copied over. I think you would still go to the home plugin as your starting point for adding and setting it up. Looking at other plugins with `*-react` they seem to be just letting people know that they contain common components that would be shared. For example: https://github.com/backstage/backstage/tree/master/plugins/search-react or https://github.com/backstage/backstage/tree/master/plugins/techdocs-react

**Review by awanlin — CHANGES_REQUESTED** [review]
Hi @malikahmontgomery, thanks for the contribution, this is great! 🚀 

Just left a few comments for you

**Comment by awanlin** [comment]
Sorry, forgot mention, can you update the title of this PR to have a quick description of this change? Just makes it a bit easier to spot in a list of PRs. Then in the body you could put something like "Closes #17737" and that should close that issue once this gets merged. Thanks in advance! 👍

**Review by awanlin — APPROVED** [review]
Thanks again for running with this @malikahmontgomery 🚀 

This looks good to me so now you just one of the maintainers to give it a look

File: yarn.lock
Comment (malikahmontgomery) [inline yarn.lock:8261]: There is an problem with 4.1.2 version of octokit-plugin-create-pull-request as workaround hardcoded to only use 4.1.1 [link to issue](https://github.com/gr2m/octokit-plugin-create-pull-request/issues/124)

File: plugins/home-react/README.md
Comment (camilaibs) [inline plugins/home-react/README.md:1]: Are there any specific reasons not to extract [componentRenderers](https://github.com/backstage/backstage/tree/master/plugins/home/src/componentRenderers) as well? They may also be useful for other plugins

**Review by camilaibs — COMMENTED** [review]
This is a great package, thank you for extracting it!
Below are just a few minor comments 🙂

File: plugins/home/api-report.md
Comment (freben) [reply plugins/home/api-report.md:16]: A pattern we've used a bit lately is to [make a `deprecated.ts`](https://github.com/backstage/backstage/blob/4c86ec146602a27d54013b71ea41890c99073fb3/packages/backend-tasks/src/deprecated.ts#L25) that just reexports from the new location, and [exporting everything from that](https://github.com/backstage/backstage/blob/4c86ec146602a27d54013b71ea41890c99073fb3/packages/backend-tasks/src/index.ts#L24)

File: .changeset/violet-pots-push.md
Comment (freben) [reply .changeset/violet-pots-push.md:3]: Yes, we usually make the package.json version 0.0.0 and make a minor changeset for it

File: plugins/home-react/README.md
Comment (malikahmontgomery) [reply plugins/home-react/README.md:1]: They are not currently exported as far as i can tell, but can be added if there is a need

File: plugins/home-react/api-report.md
Comment (malikahmontgomery) [reply plugins/home-react/api-report.md:1]: Added home-react-*

File: plugins/home-react/package.json
Comment (malikahmontgomery) [reply plugins/home-react/package.json:14]: updated type to web-library

File: plugins/home/api-report.md
Comment (malikahmontgomery) [reply plugins/home/api-report.md:16]: refactored as suggested

File: plugins/home-react/README.md
Comment (camilaibs) [reply plugins/home-react/README.md:1]: Oh, right, I just noticed that they are exported as react extensions: here: https://github.com/backstage/backstage/blob/master/plugins/home/src/plugin.ts#L45-L73

File: plugins/home/src/deprecated.ts
Comment (camilaibs) [inline plugins/home/src/deprecated.ts:21]: Ah, I think [here](https://github.com/backstage/backstage/pull/17806#discussion_r1195663539) @freben meant import from `@backstage/plugin-home-react` and re-export with "@deprecated" annotation 🤔

Comment (kmarkow) [reply plugins/home/src/deprecated.ts]: @camilaibs I pushed the update! Please check if this is what was expected from this file.

File: plugins/home-react/README.md
Comment (kmarkow) [reply plugins/home-react/README.md:3]: Done!

**Review by freben — APPROVED** [review]
Very nice, thanks!

**Pull request merged**
